core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -66,20 +66,20 @@ case class LogReadResult(info: FetchDataInfo,
                          leaderLogEndOffset: Long,
                          fetchTimeMs: Long,
                          readSize: Int,
-                         error: Option[Throwable] = None) {
+                         exception: Option[Throwable] = None) {
 
-  def errorCode = error match {
-    case None => Errors.NONE.code
-    case Some(e) => Errors.forException(e).code
+  def error: Errors = exception match {
+    case None => Errors.NONE
+    case Some(e) => Errors.forException(e)
   }
 
   override def toString = {
     "Fetch Data: [%s], HW: [%d], leaderLogEndOffset: [%d], readSize: [%d], error: [%s]"
-            .format(info, hw, leaderLogEndOffset, readSize, error)
+            .format(info, hw, leaderLogEndOffset, readSize, error.code)
```

Comment:
I kept the existing behaviour because I wasn't sure where this `toString` was being used. Do you think it would be better to output `error` instead?